rd_ui/app/scripts/services/resources.js

```diff
@@ -229,26 +255,18 @@
           }
         });
 
-        var addPointToSeries = function (seriesName, point) {
-          if (series[seriesName] == undefined) {
-            series[seriesName] = {
-              name: seriesName,
-              type: 'column',
-              data: []
-            }
-          }
-
-          series[seriesName]['data'].push(point);
-        }
-
         if (seriesName === undefined) {
           _.each(yValues, function (yValue, seriesName) {
-            addPointToSeries(seriesName, {'x': xValue, 'y': yValue});
-          });
-        } else {
-          addPointToSeries(seriesName, point);
+            this._addPointToSeries({'x': xValue, 'y': yValue}, series, seriesName);
```

Comment:
You need to use `this._addPointToSeriesIfInDateRange` (if applicable) here too, otherwise when charting a query like "SELECT x, y" it won't apply the date filter.